boot/bootutil/src/loader.c

```diff
@@ -643,16 +646,14 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
         BOOT_LOG_ERR("Image in the %s slot is not valid!",
                      (slot == BOOT_PRIMARY_SLOT) ? "primary" : "secondary");
 #endif
-        rc = 1;
+        fih_rc = fih_int_encode(1);
         goto out;
     }
 
-    /* Image in the secondary slot is valid. */
-    rc = 0;
```

Comment:
shouldn't a success code be encoded to `fih_rc`?

Reply:
You are right, and `fih_int_encode()` does exactly this: Converting an int to fih_int. You can check: fault_injection_hardening.h:203

Reply:
I see` line 637:  FIH_CALL(boot_image_check, fih_rc, state, hdr, fap, bs);` may set `fih_rc` to FIH_SUCCESS.

So this function now returns either `FIH_FAILURE` (previous `-1`), `FIH_SUCCESS` (previous `0`), and the value which is encoded by fih_int_encode(1) call (previous `1`).

I can't see initialization for either `FIH_FAILURE` and `FIH_SUCCESS`, so what the warranty that all value listed here are distinguishable?

Reply:
FIH_FAILURE and FIH_SUCCESS are just normal integer values if MCUBOOT_FIH_PROFILE is OFF. Actually their value is 0 and 1 in this case. Complex value are used only if some level of FI protection is applied.  In this case might the returns values are colliding: FIH_FAILURE  and 1 (no bootable image). will double check.